ros/src/twist_controller/pid.py

```diff
@@ -11,27 +11,27 @@ def __init__(self, kp, ki, kd, mn=MIN_NUM, mx=MAX_NUM):
         self.min = mn
         self.max = mx
 
-        self.int_val = self.last_int_val = self.last_error = 0.
+        self.int_val = 0.0
+        self.last_error = 0.0
 
     def reset(self):
         self.int_val = 0.0
-        self.last_int_val = 0.0
+        self.last_error = 0.0
 
     def step(self, error, sample_time):
-        self.last_int_val = self.int_val
-
         integral = self.int_val + error * sample_time
         derivative = (error - self.last_error) / sample_time
 
-        y = self.kp * error + self.ki * self.int_val + self.kd * derivative
-        val = max(self.min, min(y, self.max))
+        val = self.kp * error + self.ki * self.int_val + self.kd * derivative
```

Comment:
why remove max/min?

Reply:
Because it's doing the same as the `if-else` below. And that `if-else` is also taking care of not updating the integral part if we exceed actuator limits. So if we keep the `max/min` line, the `if-else` condition below is always fulfilled and therefore we keep updating the integral part, probably saturating the output always.

Reply:
i'm worried about how I in PID is handled here.  I understand the basic logic from class for getting car to center of lane but with throttle/break.. maybe there should not be correction of bias so maybe 0 for 'ki' is correct after all

Reply:
discard my comment about i in PID. should be 0